conn/bind_windows.go

```diff
@@ -321,6 +321,11 @@ func (bind *WinRingBind) Close() error {
 	return nil
 }
 
+func (bind *WinRingBind) BatchSize() int {
+	// TODO: implement batching in and out of the ring
```

Comment:
Presumably all those '!fixup' commits are intended to be squashed down, and this whole PR made into one big commit. That's fine, so long as it's complete. But it looks like the Windows part still isn't done.  So I'm wondering what the intention is with regards to that.

Reply:
Yeah, the fixups are essentially from a `git commit --fixup` workflow, there are many ways to cut this pie, but one of the slightly more helpful patterns here is that subsequent reviews can easily look at a diff of the changes made in response to comments, rather than seeking them out of a new singular whole. `git rebase --autosquash` will flatten these down, and this is the default behavior of "rebase and merge" on Github. It is our intent to have the fixups squashed.

Regarding Windows, we're not planning to include full support in this change. As you can see, we moved the API over for consistency and in anticipation of doing that. We have identified additional changes that need to be made and tested independently in order to get there, and we've started looking more deeply into those details already, but it's worthwhile getting these changes reviewed and landed in the meantime. We certainly don't want to leave Windows behind - but it has it's own ABI and API differences for offloads, and we also likely need to make some changes in upstream repos as well (e.g. to implement ReadBatch/WriteBatch support in x/* (for bindstd)), and for full effect also some changes in Wintun we don't know the full shape of yet. There may be interesting discussions to be had as to whether rio is worthwhile once batching and segment offloading are implemented, but we can have that discussion once we have more concrete change proposals and benchmark results.

Reply:
Okay. In that case, please separate Linux into its own commit, since Windows (and other platforms) will be a standalone commit too.

Reply:
That requires going back on the prior discussion we had of unifying the IO interface, so we'd have to reintroduce having two different IO APIs (one for a batch of bufs, the other for a single set of bufs). I'm not sure the split would be worthwhile, it makes the interfaces substantially more complex to read and use, and may even reduce performance due to the need for more dynamic interface usage.

Reply:
It doesn't require that. Rather, in the initial commit, you do for Linux what you did for Windows - batchsize=1 with the new api. Then in the subsequent one, you wireup sendmmsg. Will make bisecting easier too.

Reply:
The first commit in the series excludes mmsg and TSO/GRO, batchSize=1 everywhere. https://github.com/WireGuard/wireguard-go/pull/64/commits/d115be407969105d324a0ba708a6861989b15b81

Reply:
The second commit in the series implements mmsg and TSO/GRO https://github.com/WireGuard/wireguard-go/pull/64/commits/e6bfbdf72b54317c50574fcbb9ca6409c9b7b2c5